src/hmac.rs

```diff
@@ -79,3 +82,29 @@ where
     let keypair = K::from(secret_key);
     Ok(keypair)
 }
+
+////////////////////////////////////////////////////////////////////////
+/// HMAC-SHA256 based functions
+
+pub fn hmac(key: &[u8], message: &[u8]) -> Digest {
+    let mut hash = Hmac::<sha2::Sha256>::new_from_slice(key).unwrap();
+    hash.update(message);
+    let output: [u8; DIGEST_LEN] = hash.finalize().into_bytes().as_slice().try_into().unwrap();
+    Digest::new(output)
+}
+
+pub fn hkdf(
+    ikm: &[u8],
```

Comment:
Ditto. The same applies to `ikm`. As this is a secret, do we handle it as a [u8] or create a `Seed` struct around it? We did something similar for signature seeds in Sui, see https://github.com/MystenLabs/sui/blob/main/crates/sui-types/src/signature_seed.rs

Reply:
Introduced HkdfIkm for this